pkg/chunk/local/boltdb_index_client.go

```diff
@@ -117,41 +119,45 @@ func (b *boltIndexClient) Stop() {
 	b.wait.Wait()
 }
 
-func (b *boltIndexClient) NewWriteBatch() chunk.WriteBatch {
+func (b *BoltIndexClient) NewWriteBatch() chunk.WriteBatch {
 	return &boltWriteBatch{
 		tables: map[string]map[string][]byte{},
 	}
 }
 
-func (b *boltIndexClient) getDB(name string) (*bbolt.DB, error) {
+func (b *BoltIndexClient) GetDB(name string, operation int) (*bbolt.DB, error) {
```

Comment:
The refactoring here looks going beyond exporting `BoltIndexClient`. Why do we need such changes? Can you share more details, please?

Reply:
This is related to flushing boltdb feature. So the idea is to embed boltdb client from cortex into new client that I am creating in Loki and I will call exported QueryDb method for each db that I need to query i.e ones downloaded from the archive and the local one.